tensorflow/lite/micro/kernels/reduce.cc

```diff
@@ -117,6 +173,86 @@ TfLiteStatus EvalMean(TfLiteContext* context, TfLiteNode* node) {
   }
   return kTfLiteOk;
 }
+
+template <typename T>
+TfLiteStatus EvalLogic(TfLiteContext* context, TfLiteNode* node, T init_value,
+                       T reducer(const T current, const T in)) {
+  const TfLiteTensor* input = GetInput(context, node, 0);
+  const TfLiteTensor* axis = GetInput(context, node, 1);
+  TfLiteTensor* output = GetOutput(context, node, 0);
+  TF_LITE_ENSURE_TYPES_EQ(context, input->type, output->type);
+  TfLiteReducerParams* params =
+      reinterpret_cast<TfLiteReducerParams*>(node->builtin_data);
+  OpData* op_data = static_cast<OpData*>(node->user_data);
+
+  // Interpret an axis tensor with null dimensions as a scalar
+  int num_axis;
+  if (axis->dims == nullptr) {
+    num_axis = 1;
+  } else {
+    num_axis = static_cast<int>(NumElements(axis));
+  }
```

Comment:
this should become a helper function in an anonymous namespace at the top of this file.